internal/agent/reporter.go

```diff
@@ -20,7 +20,7 @@ type Sender interface {
 }
 
 // SenderFn is a function type for sending reports.
-type SenderFn func(ctx context.Context, from string, to []string, subject, body string, attachments []string) error
+type SenderFn func(ctx context.Context, from string, to []string, cc []string, bcc []string, subject, body string, attachments []string) error
```

Comment:
nitpick: this can be a bit simplified:
```suggestion
type SenderFn func(ctx context.Context, from string, to, cc, bcc []string, subject, body string, attachments []string) error
```